### PR TITLE
Add basic release notes description

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this:

/kind documentation
#### What this PR does / why we need it:
This adds a high level overview about the release notes process in SIG
release. Target is to onboard new members faster and use it as resource
for documentation purposes.

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:
cc @LappleApple

### CODE REVIEW

File: release-engineering/release-notes.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+creates drafted Markdown and JSON versions of the ongoing release. The release
+notes website as well as this repository are getting updated during the whole
+cycle.
+
+The **Branch Managers** of the Release Engineering sub-team are updating the
+main [Kubernetes repository](https://git.k8s.io/kubernetes) in parallel to the
+ongoing process. This happens usually during the actual release cut in an
+automated fashion.
+
+The **Documentation** sub-team takes care that the official
```
Comment (justaugustus) [inline release-engineering/release-notes.md:20]:
```suggestion
The **Documentation** sub-team of the Release Team ensures that the official
```

File: release-engineering/release-notes.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+[Kubernetes.io](https://kubernetes.io) website gets updated after the final
```
Comment (justaugustus) [inline release-engineering/release-notes.md:21]:
```suggestion
[kubernetes.io](https://kubernetes.io) website gets updated after the final
```

File: release-engineering/release-notes.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+release version has been cut.
+
+Patch releases can happen every time during the release and will follow its own
+schedule. The **Patch Release Team** of the Release Engineering sub-team takes
```
Comment (justaugustus) [inline release-engineering/release-notes.md:25]:
```suggestion
schedule. The **Patch Release Team** (part of the [Release Managers group](/release-managers.md)) takes
```

File: release-engineering/release-notes.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+care of updating the repository and the website by their own.
```
Comment (justaugustus) [inline release-engineering/release-notes.md:26]:
```suggestion
care of updating the relevant repositories during a patch release.
```

File: release-engineering/release-notes.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+The **Branch Managers** of the Release Engineering sub-team are updating the
+main [Kubernetes repository](https://git.k8s.io/kubernetes) in parallel to the
+ongoing process. This happens usually during the actual release cut in an
+automated fashion.
+
+The **Documentation** sub-team takes care that the official
+[Kubernetes.io](https://kubernetes.io) website gets updated after the final
+release version has been cut.
+
+Patch releases can happen every time during the release and will follow its own
```
Comment (justaugustus) [inline release-engineering/release-notes.md:24]:
```suggestion
Patch releases happen on their own [cadence](/releases/patch-releases.md), separate from the active release
```

File: release-engineering/release-notes.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+The process around the creation of the Release Notes in Kubernetes falls into
+the responsibility of SIG Release. Multiple sub-teams are involved in the
+process as well as different publishing targets (or consumers). The current
+state of the involved participants and tooling looks like this:
+
+![release notes overview](img/kubernetes-release-notes.svg "Overview about the Kubernetes Release Notes")
+
+During the release cycle, the **Release Notes Team** of SIG Release continuously
+creates drafted Markdown and JSON versions of the ongoing release. The release
+notes website as well as this repository are getting updated during the whole
```
[the remark this answers is not in the document]
Comment (saschagrunert) [reply release-engineering/release-notes.md]: Added :+1:

**Comment by saschagrunert** [comment]
> @saschagrunert -- Thanks for adding this! I've made some language suggestions.

Thank you for the detailed review! I applied the suggestions and added the link to the release notes website.

**Comment by justaugustus** [comment]
Thanks again, @saschagrunert!
/lgtm
/approve

**Pull request merged**